var/spack/repos/builtin/packages/openblas/package.py

```diff
@@ -114,6 +89,10 @@ class Openblas(MakefilePackage):
     patch('openblas_fujitsu.patch', when='%fj')
 
     conflicts('%intel@16', when='@0.2.15:0.2.19')
+    conflicts('~shared~static')
+
+    # Use newer assembler to allow avx2 instructions
+    depends_on('binutils', type='build', when='os=rhel6')
```

Comment:
See https://github.com/spack/spack/issues/8852 about difficulties configuring newer GCC to use newer binutils (the default Spack GCC installation uses system binutils), otherwise I'd agree with you about proper compiler configuration. This is the most straightforward way I can see to avoid https://github.com/spack/spack/issues/12235, which I independently encountered when building openblas on an older system.

Reply:
My official stance on binutils is here: https://spack.readthedocs.io/en/latest/getting_started.html#binutils

Basically, binutils is needed by every package in Spack, so it would be overkill to make it a dependency of every package.

Reply:
OK, reverting so it uses the system binutils (or whatever the compiler provides) even though it'll probably generate failures on RHEL6 builds.